### PR TITLE
change snappyOutputStream  writeHeader order to fix when outputStream…

### PR DESCRIPTION
**Pull request by Neilxzn**
In version 1.1.2.x, SnappyOutputStream will generate a snappy header no matter if it has been written.
But In master branch,  SnappyOutputStream will not generate a header if it has not been written. So to fix it, I change some code in  SnappyOutputStream.java. After fixed it, the code  will have the same behavior in version 1.1.2.x and 1.1.7.x.
```
        ByteArrayOutputStream b = new ByteArrayOutputStream();
        SnappyOutputStream os = new SnappyOutputStream(b);
        os.close();
        int len= b.size();
        assertEquals(16,len);
```

### CODE REVIEW

File: src/main/java/org/xerial/snappy/SnappyOutputStream.java [hunk 1]
```diff
@@ -365,15 +365,21 @@ protected void dumpOutput()
...
         if (!headerWritten) {
             outputCursor = writeHeader();
             headerWritten = true;
         }
 
+        if (inputCursor <= 0) {
+            return; // no need to dump
+        }
+
+//        if (!headerWritten) {
```
Comment (xerial) [inline src/main/java/org/xerial/snappy/SnappyOutputStream.java]: Please remove these comments as we already have the same code block in the above

File: src/main/java/org/xerial/snappy/SnappyOutputStream.java [hunk 2]
```diff
@@ -365,15 +365,16 @@ protected void dumpOutput()
...
         if (!headerWritten) {
             outputCursor = writeHeader();
             headerWritten = true;
         }
 
+        if (inputCursor <= 0) {
+            return; // no need to dump
+        }
+
+
```
Comment (xerial) [inline src/main/java/org/xerial/snappy/SnappyOutputStream.java]: Remove this extra blank line

**Comment by Neilxzn** [comment]
thank you for your review @xerial

**Comment by xerial** [comment]
@Neilxzn Thanks for the fix. I'm now trying to remember the past context about this behavior. If no data is compressed, producing no data might be preferable for some applications (e.g., Spark). 

@JoshRosen might have remembered the expected behavior.

**Comment by Neilxzn** [comment]
@xerial  I found this problem when I use Spark 2.4. [https://issues.apache.org/jira/browse/SPARK-27267](SPARK-27267)

**Comment by xerial** [comment]
https://issues.apache.org/jira/browse/SPARK-27267

**Pull request merged**